lib/slim/embedded.rb

```diff
@@ -153,9 +153,11 @@ class SassEngine < TiltEngine
 
       def tilt_render(tilt_engine, tilt_options, text)
         text = tilt_engine.new(tilt_options.merge(
-          style: options[:pretty] ? :expanded : :compressed,
-          cache: false)) { text }.render
-        text = text.chomp
+          style: options[:pretty] ? :expanded : :compressed
+        ).merge!(
+          tilt_engine <= ::Tilt::SassTemplate && tilt_engine::Engine ? { cache: false } : { charset: false }
+        )) { text }.render
+        text = text.chomp unless text.frozen?
```

Comment:
sass-embedded returns a frozen string created from protobuf that does not have a new line at the end. It is more performant to not duplicate the frozen string.

Reply:
Is this necessary?

Reply:
The alternative would be `(+text).chomp`, which is always slower.